plugins/woocommerce/includes/admin/helper/class-wc-helper-subscriptions-api.php

```diff
@@ -136,9 +164,83 @@ public static function deactivate( $request ) {
 			wp_send_json_error(
 				array(
 					'message' => __( 'There was an error deactivating your subscription. Please try again.', 'woocommerce' )
-				)
+				),
+				400
+			);
+		}
+	}
+
+	/**
+	 * Install a WooCommerce.com produc
+	 */
+	public static function install( $request ) {
+		$product_key = $request->get_param('product_key');
+		$subscriptions = WC_Helper::get_subscription( $product_key );
+
+		if ( empty( $subscriptions ) ) {
+			wp_send_json_error(
+				array(
+					'message' => __( 'We couldn\'t find this subscription.', 'woocommerce' )
+				),
+				404
+			);
+		}
+
+		if ( $subscriptions['expired'] === true ) {
+			wp_send_json_error(
+				array(
+					'message' => __( 'This subscription has expired.', 'woocommerce' )
+				),
+				402
 			);
 		}
+
+		if ( $subscriptions['maxed'] === true ) {
+			wp_send_json_error(
+				array(
+					'message' => __( 'All licenses for this subscription are already in use.', 'woocommerce' )
+				),
+				402
+			);
+		}
```

Comment:
These checks _may be_ excessive as the similar validation are implemented in other places:
- Installation_Manager that is triggered below will check for an active subscription before starting installation, and trigger an error if it's not found. 
https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/wccom-site/installation/installation-steps/class-wc-wccom-site-installation-step-get-product-info.php#L108-L110
- On activate request, WCCOM will check whether the subscription is not expired and has remaining licenses.

Reply:
Thanks @denho! I will update the checks to remove duplication.